### PR TITLE
use interface in error message when check instance

### LINKED ISSUE #3375: class used in error messages instead of interface
How to reproduce:

```
data EmptyU = VOID

data Unifies : a -> b -> Type where
  ItUnifies : Unifies a b

interface DecUnifies a where
  decUnifies : a -> a -> Dec (Unifies a a)

DecUnifies EmptyU where
  decEq VOID VOID = Yes ItUnifies
```

Expected behaviour:
```
/home/arian/Projects/thesis2/src/Servis/API.idr:24:1:
Servis.API.decEq not a method of interface DecUnifies

```

Current behaviour:
```
/home/arian/Projects/thesis2/src/Servis/API.idr:24:1:
Servis.API.decEq not a method of class DecUnifies

```

### PR DESCRIPTION
**Pull request by zjhmale**
Fixes #3375

### CODE REVIEW

File: src/Idris/Parser/Helpers.hs [hunk 1]
```diff
@@ -276,7 +276,7 @@ idrisStyle = IdentifierStyle {
       ["let", "in", "data", "codata", "record", "corecord", "Type", "do", "dsl",
       "import", "impossible", "case", "of", "total", "partial", "mutual",
       "infix", "infixl", "infixr", "rewrite", "where", "with", "syntax",
-      "proof", "postulate", "using", "namespace", "class", "instance",
+      "proof", "postulate", "using", "namespace", "instance",
```
Comment (zjhmale) [inline src/Idris/Parser/Helpers.hs]: i'm not quit sure that we should remove `class` from reserved words.

Comment (ahmadsalim) [inline src/Idris/Parser/Helpers.hs]: We should not yet. While it is deprecated now and people should change from it, I don't think that this is the right PR to break support.

File: src/Idris/Core/TT.hs [hunk 2]
```diff
@@ -1674,7 +1674,7 @@ pureTerm (Bind n b sc) = notClassName n && pureBinder b && pureTerm sc where
     pureBinder (Let t v) = pureTerm t && pureTerm v
     pureBinder t = pureTerm (binderTy t)
 
-    notClassName (MN _ c) | c == txt "__class" = False
+    notClassName (MN _ c) | c == txt "__interface" = False
```
Comment (ahmadsalim) [inline src/Idris/Core/TT.hs]: Maybe this should be `notInterfaceName` while you were changing it?

File: src/Idris/Elab/Class.hs [hunk 3]
```diff
@@ -164,12 +164,12 @@ elabClass info syn_in doc fc constraints tn tnfc ps pDocs fds ds mcn cd
     checkDefaultSuperclassInstance :: PDecl -> Idris ()
     checkDefaultSuperclassInstance (PInstance _ _ _ fc cs _ _ _ n _ ps _ _ _ _)
         = do when (not $ null cs) . tclift
-                $ tfail (At fc (Msg "Default superclass instances can't have constraints."))
+                $ tfail (At fc (Msg "Default superinterface instances can't have constraints."))
```
Comment (ahmadsalim) [inline src/Idris/Elab/Class.hs]: I am not sure "superinterface" is a valid English word. Perhaps use "super-interface" or "super interface" instead?

**Comment by ahmadsalim** [comment]
So the tests failed because some of them are dependent on "__class" and the old messages that relied on that. You should probably change the expected files as well.

File: src/Idris/Elab/Class.hs (hunk 3 above)
Comment (zjhmale) [inline src/Idris/Elab/Class.hs]: cool

File: src/Idris/Core/TT.hs (hunk 2 above)
Comment (zjhmale) [inline src/Idris/Core/TT.hs]: i just thought to change all the `class` in strings in this PR and all the `class` in docs and code level next PRs respectively.

**Comment by zjhmale** [comment]
@ahmadsalim @melted @Melvar please review.

**Comment by ahmadsalim** [comment]
@zjhmale I think it looks fine so far! Let us see how CI behaves.

**Comment by zjhmale** [comment]
@ahmadsalim cool

**Comment by zjhmale** [comment]
also update some  `instance` to `implementation` in string literal.

**Comment by ahmadsalim** [comment]
@zjhmale There are some conflicts now, would you kindly take a look at them?

**Comment by zjhmale** [comment]
@ahmadsalim definitely!

**Comment by zjhmale** [comment]
poke @ahmadsalim

**Comment by ahmadsalim** [comment]
@zjhmale I think you need to update the `ibcVersion` variable in `IBC.hs` since you changed the entry names, but otherwise it looks good.

**Comment by zjhmale** [comment]
@ahmadsalim bump up ibc version to 149 😄

**Comment by ahmadsalim** [comment]
Thanks!

**Pull request merged**